src/document/crdt/tree.ts

```diff
@@ -452,74 +597,63 @@ export class CRDTTree extends CRDTGCElement {
-      throw new Error(`cannot find node at ${pos}`);
-    }
-
-    // Find the appropriate position. This logic is similar to the logical to
-    // handle the same position insertion of RGA.
-    let current = treePos;
-    while (
-      current.node.next?.pos.getCreatedAt().after(editedAt) &&
-      current.node.parent === current.node.next.parent
-    ) {
-      current = {
-        node: current.node.next,
-        offset: current.node.next.size,
-      };
-    }
-
-    const right = this.indexTree.findPostorderRight(treePos)!;
-    return [current, right];
-  }
-
-  /**
-   * `findTreePosWithSplitText` finds `TreePos` of the given `CRDTTreePos` and
+   * `findNodesAndSplitText` finds `TreePos` of the given `CRDTTreeNodeID` and
    * splits the text node if necessary.
    *
-   * `CRDTTreePos` is a position in the CRDT perspective. This is
+   * `CRDTTreeNodeID` is a position in the CRDT perspective. This is
    * different from `TreePos` which is a position of the tree in the local
    * perspective.
    */
-  public findTreePosWithSplitText(
+  public findNodesAndSplitText(
     pos: CRDTTreePos,
     editedAt: TimeTicket,
-  ): [TreePos<CRDTTreeNode>, CRDTTreeNode] {
-    const treePos = this.toTreePos(pos);
-    if (!treePos) {
+  ): [CRDTTreeNode, CRDTTreeNode] {
+    const treeNodes = this.toTreeNodes(pos);
+
+    if (!treeNodes) {
       throw new Error(`cannot find node at ${pos}`);
     }
+    const [parentNode] = treeNodes;
+    let [, leftSiblingNode] = treeNodes;
 
     // Find the appropriate position. This logic is similar to the logical to
     // handle the same position insertion of RGA.
-    let current = treePos;
-    while (
-      current.node.next?.pos.getCreatedAt().after(editedAt) &&
-      current.node.parent === current.node.next.parent
-    ) {
-      current = {
-        node: current.node.next,
-        offset: current.node.next.size,
-      };
-    }
 
-    if (current.node.isText) {
-      const split = current.node.split(current.offset);
+    if (leftSiblingNode.isText) {
+      const absOffset = leftSiblingNode.pos.getOffset();
+      const split = leftSiblingNode.split(
+        pos.getLeftSiblingID().getOffset() - absOffset,
+        absOffset,
+      );
+
       if (split) {
-        this.insertAfter(current.node, split);
-        split.insPrev = current.node;
+        split.insPrev = leftSiblingNode;
+        this.nodeMapByPos.put(split.pos, split);
+
+        if (leftSiblingNode.insNext) {
+          leftSiblingNode.insNext.insPrev = split;
+          split.insNext = leftSiblingNode.insNext;
+        }
+        leftSiblingNode.insNext = split;
       }
     }
 
-    const right = this.indexTree.findPostorderRight(treePos)!;
-    return [current, right];
+    const index =
+      parentNode === leftSiblingNode
+        ? 0
+        : parentNode.allChildren.indexOf(leftSiblingNode) + 1;
+
+    for (let i = index; i < parentNode.allChildren.length; i++) {
+      const next = parentNode.allChildren[index];
```

Comment:
```suggestion
      const next = parentNode.allChildren[i];
```
Is this intended? or just a typo?

Reply:
oh it's typo,

Reply:
It would be good to add a test case for this together.